### PR TITLE
Corrected RegNet implementation

### PR DESCRIPTION
**Pull request by AdityaKane2001**
@LukeWood @sayakpaul 

Corrected RegNets implementation which was causing tests to fail:
- Added input_tensor argument.
- `num_classes` -> `classes`

### CODE REVIEW

File: keras_cv/models/regnet.py [hunk 1]
```diff
@@ -0,0 +1,1543 @@
...
+        )
+
+    img_input = utils.parse_model_inputs(input_shape, input_tensor)
+    x = img_input
+
+    if include_rescaling:
+        x = layers.Rescaling(scale=1.0 / 255.0)(x)
+    x = Stem(name=model_name)(x)
+
+    in_channels = 32  # Output from Stem
```
Comment (sayakpaul) [inline keras_cv/models/regnet.py:763]: Can't it be programmatically calculated?

File: keras_cv/models/regnet.py [hunk 2]
```diff
@@ -0,0 +1,1543 @@
...
+
+    for stage_index in range(4):
```
Comment (sayakpaul) [inline keras_cv/models/regnet.py:765]: Prefer assigning the magic number to a meaningful variable.

File: keras_cv/models/regnet.py [hunk 3]
```diff
@@ -0,0 +1,1543 @@
...
+            depth,
+            group_width,
+            in_channels,
+            out_channels,
+            name=model_name + "_Stage_" + str(stage_index),
+        )(x)
+        in_channels = out_channels
+
+    if include_top:
+        x = Head(classes=classes, activation=classifier_activation)(x)
```
Comment (sayakpaul) [inline keras_cv/models/regnet.py:780]: @LukeWood @ianstenbit isn't `num_classes` a better variable name? I know we have been following `classes` for models, but wanted to revisit it.

**Review by sayakpaul — CHANGES_REQUESTED** [review]

File: keras_cv/models/regnet.py (hunk 3 above)
Comment (AdityaKane2001) [reply keras_cv/models/regnet.py:780]: I agree here.

File: keras_cv/models/regnet.py (hunk 1 above)
Comment (AdityaKane2001) [reply keras_cv/models/regnet.py:763]: `x.shape[-1]`?

File: keras_cv/models/regnet.py (hunk 3 above)
Comment (LukeWood) [reply keras_cv/models/regnet.py:780]: Yeah... we have gone back and forth on this a lot of times.

@fchollet had some thoughts on this.

**Comment by AdityaKane2001** [comment]
@sayakpaul @LukeWood 

Made changes from the review.

File: keras_cv/models/regnet.py [hunk 4]
```diff
@@ -202,6 +202,8 @@
     },
 }
 
+NUM_STAGES = 4
```
Comment (LukeWood) [inline keras_cv/models/regnet.py:205]: Let’s  have this adjacent to the loop as a local variable.

**Comment by LukeWood** [comment]
This is blocked on refactoring model tests to be written as a file per model.  Currently this will make our tests OOM.

**Pull request closed without merging**